phys2bids/utils.py

```diff
@@ -23,15 +23,13 @@ def check_input_ext(filename, ext):
     """
     Checks that the given file has the given extension
     """
-    if '.gz' in ext:
-        if filename[-len(ext):] != ext:
-            filename = filename + ext
-        return filename
-    else:
-        if ext[0] != '.':
-            ext = '.' + ext
+    if '.gz' in filename:
+        filename = filename[:-3]
+
+    if ext[0] != '.':
+        ext = '.' + ext
```

Comment:
This is slightly preference, but I think using the string methods for these checks might be a bit cleaner:

```python
if not ext.startswith('.'):
    ext = '.' + ext
```